src/main/java/frc/trigon/robot/motorsimulation/MotorSimulation.java

```diff
@@ -0,0 +1,140 @@
+        this.positionVoltageRequest = positionVoltageRequest;
+        motionMagicRequest = null;
+        double voltage = pidController.calculate(getPosition(), positionVoltageRequest.Position);
+        setVoltage(voltage);
+    }
+
+    public void setControl(MotionMagicVoltage motionMagicRequest) {
+        this.motionMagicRequest = motionMagicRequest;
+        positionVoltageRequest = null;
+        double pidOutput = profiledPIDController.calculate(getPosition(), motionMagicRequest.Position);
+        double feedforwardOutput = calculateFeedforward(
+                config.feedForwardConfigs,
+                Units.rotationsToRadians(motionMagicRequest.Position / config.conversionFactor),
+                profiledPIDController.getGoal().velocity
+        );
+        double output = pidOutput + feedforwardOutput;
+        setVoltage(output);
+    }
+
+    public void setControl(DutyCycleOut dutyCycleRequest) {
+        double voltage = Conversions.compensatedPowerToVoltage(dutyCycleRequest.Output, voltageCompensationSaturation);
+        setVoltage(voltage);
+    }
+
+    public double getVoltage() {
+        return voltage;
+    }
+
+    public double getPosition() {
+        return getPositionRevolutions() * config.conversionFactor;
+    }
+
+    public double getVelocity() {
+        return getVelocityRevolutionsPerSecond() * config.conversionFactor;
+    }
+
+    private static void updateRegisteredSimulations() {
+        for (MotorSimulation motorSimulation : REGISTERED_SIMULATIONS) {
+            motorSimulation.updateMotor();
+            if (motorSimulation.motionMagicRequest != null)
+                motorSimulation.setControl(motorSimulation.motionMagicRequest);
+            else if (motorSimulation.positionVoltageRequest != null)
+                motorSimulation.setControl(motorSimulation.positionVoltageRequest);
+        }
+    }
+
+    private void setVoltage(double voltage) {
+        double compensatedVoltage = MathUtil.clamp(voltage, -voltageCompensationSaturation, voltageCompensationSaturation);
+        this.voltage = compensatedVoltage;
+        setInputVoltage(compensatedVoltage);
+    }
+
+    private void enablePIDContinuousInput(boolean enableContinuousInput) {
+        if (enableContinuousInput) {
+            pidController.enableContinuousInput(-(0.5 * config.conversionFactor), 0.5 * config.conversionFactor);
+            profiledPIDController.enableContinuousInput(-(0.5 * config.conversionFactor), 0.5 * config.conversionFactor);
+        } else {
+            pidController.disableContinuousInput();
+            profiledPIDController.disableContinuousInput();
+        }
+    }
+
+    /**
+     * Calculates the feedforward
+     *
+     * @param feedForwardConfiguration The feedforward configuration
+     * @param targetPositionRadians    The target position in radians
+     * @param targetVelocity           The target velocity
+     * @return The calculated feedforward voltage
```

Comment:
For our conventions, we have un-capitlized letters for javadoc params / returns. 
For the main idea (`Calculates the feedforward`), we have a period at the end of the sentence.
Mention that the target velocity is in the conversion factor